src/server/game/Entities/Item/Item.cpp

```diff
@@ -947,7 +956,15 @@ void Item::SetItemRandomProperties(ItemRandomEnchantmentId const& randomPropId)
 
 void Item::UpdateItemSuffixFactor()
 {
-    uint32 suffixFactor = GenerateEnchSuffixFactor(GetEntry());
+    if (!GetTemplate()->GetRandomSuffix())
```

Comment:
and maybe ITEM_FIELD_PROPERTY_SEED should be saved to db and this function removed?

Reply:
That would work for most cases yes. But some items (such as heirlooms) keep changing their item level, which keeps changing the seed value.

Reply:
heirlooms dont have a random suffix, so no

Reply:
Fair enough about the heirlooms. Although I'd like to do the suffix factor changes in a followup PR, with some other item creation refactors/cleanups, if that's okay?